engine/src/main/java/org/hibernate/validator/internal/engine/messageinterpolation/LocalizedMessage.java

```diff
@@ -31,10 +34,10 @@ public boolean equals(Object o) {
 
 		LocalizedMessage that = (LocalizedMessage) o;
 
-		if ( locale != null ? !locale.equals( that.locale ) : that.locale != null ) {
+		if ( message != null ? !message.equals( that.message ) : that.message != null ) {
```

Comment:
It might not be much of the value, but the way `LocalizedMessage` is used we shouldn't have `locale` or `message` as `null` and also that if statement above:
```java
if ( o == null || getClass() != o.getClass() ) {
    return false;
}
``` 
I don't remember what we've done in the previous case, did we removed them, or not?

Reply:
Ah sure. I did remove the test for the `hashCode()` but forgot to do the same for `equals()`. Will do.

Reply:
@marko-bekhta fixed thanks